test/integration/etcd_sharding_test.go

```diff
@@ -0,0 +1,133 @@
+// +build integration,etcd
+
+package integration
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/coreos/go-etcd/etcd"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	etcdtesting "k8s.io/kubernetes/pkg/storage/etcd/testing"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	"github.com/openshift/origin/pkg/cmd/server/api"
+	projectapi "github.com/openshift/origin/pkg/project/api"
+	testutil "github.com/openshift/origin/test/util"
+	testserver "github.com/openshift/origin/test/util/server"
+)
+
+func TestEtcdSharding(t *testing.T) {
+	// Start a separate etcd server
+	kubeEtcdServer := etcdtesting.NewEtcdTestClientServer(t)
+	defer kubeEtcdServer.Terminate(t)
+	kubeEtcdClient := etcd.NewClient(kubeEtcdServer.ClientURLs.StringSlice())
+
+	// Generate openshift configuration
+	openShiftMasterConfig, err := testserver.DefaultMasterOptions()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// Configure openshift to write kubernetes data to a separate etcd
+	openShiftMasterConfig.KubernetesMasterConfig.EtcdClientInfo = &api.EtcdConnectionInfo{
+		URLs: kubeEtcdServer.ClientURLs.StringSlice(),
```

Comment:
nm... I thought you were copying the config... you're creating a new one... so they're already empty